src/irmin-git/irmin_git.ml

```diff
@@ -747,70 +736,76 @@ struct
 
   type branch = B.t
 
-  type endpoint = S.Endpoint.t
+  type endpoint = Mimic.ctx * Smart_git.Endpoint.t
```

Comment:
Would it be possible to hide `Mimic.ctx` into `Smart_git.Endpoint.t`? Why does it need to be separated here?

Reply:
It's not really the same purpose and `Mimic.ctx` can be used with something else than a `Smart_git.Endpoint.t` (e.g. an `Uri.t` for HTTP/AF for example).

Reply:
ok makes sense.